### PR TITLE
Feature/edit submitted slop

### PR DESCRIPTION
**Pull request by fostert14**
## Describe your changes
In summary I added a form that allows the user to edit a submitted slop if it belongs to that user. The route is 

In summary, I added an EditSlop component that will allow the author of a submitted slop to have an Edit button appear, allowing the user to update and edit the slop they submitted.

1. In submit-list route, if the current user matches the author id, edit button will appear. If the user is an admin, approve button appears in it's place. If the current user is neither, then the button is not rendered. This logic is found in SubmitList.jsx. 
2. When clicking on the Edit button that will take you to the /edit-slop${movieId} route, which will populate the edit form with almost all the movie data. I did not include an option to edit the image since I believe that is being worked on in another ticket.
3. After clicking submit, the movie data is then updated, and the user is navigated back to the submit-list route.

I made a few other changes in order to add this component.

1. Added a GET_MOVIE query, found in src/graphql/queries/get-movie-by-id. This pulls the individual movie data instead of the list of data.
2. In get-movies.js, I added pulling author data to help render the button dynamically based on the userid.
3. Added a mutation to edit slops in src/graphql/mutations/edit-slop-submissions.
4. Edited the padding of the text form box so that it would match in heigh based on the figma layout.
5. Made changes to useCurrentUser.js to pull userId and admin privileges. 

A few things to note:
The layout for submit-list does not match the figma design as there is already a ticket/pull request for that.
As mentioned above, the edit image is not integrated due to this being in another ticket.
I receive a warning when in the EditSlop component from the combobox

## Issue ticket number and link
Slop-17 Edit Submitted Slop

https://tripleten-apiary.atlassian.net/browse/SLOP-17?atlOrigin=eyJpIjoiOTU0MGNkZGMzMGYzNGI2N2I1MzFiNzRlNDJiOGJhZmQiLCJwIjoiaiJ9

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/components/EditSlop.jsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  };
+
+  const handleInputChange = (fieldname) => (event) => {
+    setSlopData({ ...slopData, [fieldname]: event.target.value });
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    //parse all numerical values to be accepted when submitting
```
Comment (Dillona25) [inline src/components/EditSlop.jsx:63]: Great way to do this!

File: src/components/EditSlop.jsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    const releaseYearInt = parseInt(slopData.releaseYear, 10);
```
Comment (Dillona25) [inline src/components/EditSlop.jsx:64]: Great way to do this and great comments!

File: src/components/SubmitList.jsx [hunk 3]
```diff
@@ -1,7 +1,36 @@
...
+
+  //edit function for dynamic button
+  const handleEdit = (movieId) => {
+    navigate(`/edit-slop/${movieId}`);
+  };
+
+  //dynamic button for handleEdit. If the author id matches the user, Edit Button appears.
+  // if the current user is an admin, approve button appears.
+  // if neither of these conditions are met, the button does not appear.
+  const renderButton = (movie) => {
```
Comment (Dillona25) [inline src/components/SubmitList.jsx:23]: Great conditional statement here

**Review by Dillona25 — APPROVED** [review]
Code all looks good. Task was appropriately addressed without errors on my end.

**Review by jahorwitz — APPROVED** [review]

**Pull request merged**